### PR TITLE
Isolating sstable and stacker in independant crates.

### PR DESCRIPTION
**Pull request by fulmicoton**
Both crate will be used in the new (optional + dynamic) fastfield work.

### CODE REVIEW

File: stacker/src/arena_hashmap.rs [hunk 1]
```diff
@@ -41,15 +40,16 @@ impl KeyValue {
...
-/// stored in a user defined memory arena.
+/// Its main particularity is that rather than storing its
+/// keys in the heap, keys are stored in a memory arena
+/// inline with the values.
 ///
 /// The quirky API has the benefit of avoiding
 /// the computation of the hash of the key twice,
 /// or copying the key as long as there is no insert.
-pub struct TermHashMap {
+pub struct ArenaHashMap {
```
Comment (PSeitz) [inline stacker/src/arena_hashmap.rs:52]: I think the new name is too generic, the implementation is very specific, mapping &[u8] keys to ids

File: stacker/src/arena_hashmap.rs [hunk 2]
```diff
@@ -153,22 +158,30 @@ impl TermHashMap {
         }
     }
 
-    fn set_bucket(&mut self, hash: u32, key_value_addr: Addr, bucket: usize) -> UnorderedTermId {
+    #[inline]
+    fn set_bucket(&mut self, hash: u32, key_value_addr: Addr, bucket: usize) -> UnorderedId {
         self.occupied.push(bucket);
-        let unordered_term_id = self.len as UnorderedTermId;
+        let unordered_term_id = self.len as UnorderedId;
```
Comment (PSeitz) [inline stacker/src/arena_hashmap.rs]:
```suggestion
        let unordered_id = self.len as UnorderedId;
```

File: stacker/src/arena_hashmap.rs [hunk 3]
```diff
@@ -153,22 +158,30 @@ impl TermHashMap {
...
         self.len += 1;
         self.table[bucket] = KeyValue {
             key_value_addr,
             hash,
-            unordered_term_id,
+            unordered_id: unordered_term_id,
```
Comment (PSeitz) [inline stacker/src/arena_hashmap.rs]:
```suggestion
            unordered_id,
```

File: stacker/src/arena_hashmap.rs (hunk 1 above)
Comment (fulmicoton) [reply stacker/src/arena_hashmap.rs:52]: Yeah... So `Term` will soon be incorrect, as I use the stacker with field names as keys in dynamic fast fields.

I could have something like `ArenaBytesHashMap` or something like this, but I think at this point `ArenaHashMap` is ok.

**Pull request merged**